integration-tests/oidc-tenancy/src/main/java/io/quarkus/it/keycloak/TenantResource.java

```diff
@@ -105,6 +107,22 @@ public String userNameServiceNoDiscovery(@PathParam("tenant") String tenant) {
         return userNameService(tenant, false);
     }
 
+    @AuthorizationCodeFlow
+    @GET
+    @Path("code-flow-auth-mech-annotation")
+    @RolesAllowed("user")
+    public String codeFlowAuthMechSelectedExplicitly() {
+        return "code-flow-auth-mech-annotation";
```

Comment:
Minor suggestion - here it can make sense to verify the injected IdToken returns some name from getName(), and in the bearer test - that IdToken getName() returns null

Reply:
I'd like to propose to drop `@RolesAllowed` in at least one of these tests, to confirm that adding `@Authenticated` is not necessary

Reply:
@michalvavrik Please ignore my last comment - I see it is tested in the other test

Reply:
> I'd like to propose to drop @RolesAllowed in at least one of these tests, to confirm that adding @Authenticated is not necessary

Thanks. Yes, I remember trying this extensively, though truth be told, memory is bit vague now.

> Minor suggestion - here it can make sense to verify the injected IdToken returns some name from getName(), and in the bearer test - that IdToken getName() returns null

will do

Reply:
Done.